libs/ledger-live-common/src/hw/ftsLoadImage.ts

```diff
@@ -0,0 +1,251 @@
+                0xe0,
+                0x62,
+                0x00,
+                0x00,
+                Buffer.from([]),
+                [0x9000, 0x5501]
+              );
+
+              const commitStatus = commitResponse.readUInt16BE(
+                commitResponse.length - 2
+              );
+              const commitStatusStr = commitStatus.toString(16);
+              // reads last 2 bytes which correspond to the status
+
+              if (commitStatus === 0x5501) {
+                return subscriber.error(
+                  new ImageCommitRefusedOnDevice(commitStatusStr)
+                );
+              } else if (commitStatus !== 0x9000) {
+                return subscriber.error(
+                  new TransportError(
+                    "Unexpected device response",
+                    commitStatusStr
+                  )
+                );
+              }
+
+              subscriber.next({
+                type: "imageLoaded",
+              });
+
+              subscriber.complete();
+            }),
+            catchError((e: unknown) => {
+              if (
+                e instanceof DeviceOnDashboardExpected ||
+                (e &&
+                  e instanceof TransportStatusError &&
+                  [0x6e00, 0x6d00, 0x6e01, 0x6d01, 0x6d02].includes(
+                    // @ts-expect-error typescript not checking agains the instanceof
+                    e.statusCode
+                  ))
+              ) {
+                return from(getAppAndVersion(transport)).pipe(
+                  concatMap((appAndVersion) => {
+                    return !isDashboardName(appAndVersion.name)
+                      ? attemptToQuitApp(transport, appAndVersion)
+                      : of<LoadImageEvent>({
+                          type: "appDetected",
+                        });
+                  })
+                );
+              }
+              return throwError(e);
+            })
+          )
+          .subscribe(subscriber);
+
+        return () => {
+          timeoutSub.unsubscribe();
+          sub.unsubscribe();
+        };
+      })
+  );
+
+  return sub as Observable<LoadImageEvent>;
+}
+
+const generateFtsImageFormat: (
+  imgHex: string,
+  compressImage: boolean
+) => Promise<Buffer> = async (imgHex, compressImage) => {
+  const width = 400;
+  const height = 672;
+  const bpp = 2; // value for 4 bits per pixel
+  const compression = compressImage ? 1 : 0;
+
+  const header = Buffer.alloc(8);
+
+  header.writeUInt16LE(width, 0); // width
+  header.writeUInt16LE(height, 2); // height
+  header.writeUInt8((bpp << 4) | compression, 4);
+
+  const imgData = Buffer.from(imgHex, "hex");
+
+  if (!compressImage) {
+    const dataLength = imgData.length;
+    header.writeUInt8(dataLength & 0xff, 5); // lowest byte
+    header.writeUInt8((dataLength >> 8) & 0xff, 6); // middle byte
+    header.writeUInt8((dataLength >> 16) & 0xff, 7); // biggest byte
```

Comment:
we're doing this because `dataLength` is too big and we need to write it on 3 bytes ?